0003-minimal-snarkyjs-checks-part1.md

```diff
@@ -0,0 +1,91 @@
+## Summary
+[summary]: #summary
+
+A minimal layer of SnarkyJS checks that won't break with typical protocol development.
+
+## Motivation
+[motivation]: #motivation
+
+Objective: Empower the protocol development to continue swiftly, while preventing maximally complex SnarkyJS bitrot. Out-of-scope for this RFC: re-enabling the rest of those tests in SnarkyJS CI, and changing the submodule layout within Mina.
+
+Background: Currently, SnarkyJS and, transitively, SnarkyJS-bindings is included in the main Mina repo as a submodule. Until recently, extensive CI jobs ensured that SnarkyJS remains correct as Mina evolves. Unfortunately, this setup has many issues: (1) Mina is owned by the Mina Foundation and SnarkyJS is owned by O(1) Labs; it's semantically awkward to rely on Mina's CI for testing SnarkyJS, (2) the dependancy graph feels a bit inverted; as SnarkyJS is a client of the Mina daemon's internals rather than Mina daemon's internals being a client of SnarkyJS, and most importantly (3) it's causing velocity issues with protocol development. Why? Many parts of SnarkyJS is tightly coupled to zkApp representations and behavior -- this means that even trivial renaming or reordering of fields requires SnarkyJS changes (albeit in a semi-automatic fashion).
+
+Having said that, there are important reasons to keep some tests running on Mina's PRs: For example, many cryptography changes have the potential to cause proof generation on SnarkyJS, and by far the best place to address that is the PR introducing the cryptography change.
+
+As of this RFC, all the SnarkyJS are configured to "soft-fail" on Mina's CI; this was a quickfix to empower quick Mina development, but is creating opportunities for bitrot in these situations so this RFC aims to quickly reconcile this temporary situation.
+
+Success Metric: Protocol developers are undisturbed by SnarkyJS tests in day-to-day development, but when deep changes to cryptography are taken, 
```

Comment:
```suggestion
Success Metric: Protocol developers are undisturbed by SnarkyJS tests in day-to-day development. However, when deep changes to cryptography are made, the following requirements apply.
```

Reply:
something like this to introduce the requirements (I think we mean the requirements apply only to deep changes)